modules/prometheus/server/src/main/scala/MonitoringServerInterceptor.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017-2018 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package freestyle.rpc
+package prometheus
+package server
+
+import java.time.{Clock, Instant}
+
+import freestyle.rpc.interceptors.GrpcMethodInfo
+import freestyle.rpc.prometheus.shared.Configuration
+import io.grpc._
+
+case class MonitoringServerInterceptor(clock: Clock)(implicit CFG: Configuration)
+    extends ServerInterceptor {
+
+  val serverMetrics: ServerMetrics = ServerMetrics(CFG)
+
+  override def interceptCall[Req, Res](
+      call: ServerCall[Req, Res],
+      requestHeaders: Metadata,
+      next: ServerCallHandler[Req, Res]): ServerCall.Listener[Req] = {
+
+    val method: MethodDescriptor[Req, Res] = call.getMethodDescriptor
+    val metricsMethod: ServerMetricsForMethod =
+      ServerMetricsForMethod(GrpcMethodInfo(method), serverMetrics)
+
+    val monitoringCall: MonitoringServerCall[Req, Res] =
+      MonitoringServerCall[Req, Res](call, clock, metricsMethod)
+
+    MonitoringServerCallListener[Req](next.startCall(monitoringCall, requestHeaders), metricsMethod)
+  }
+}
+
+object MonitoringServerInterceptor {
+
+  def apply(implicit CFG: Configuration): MonitoringServerInterceptor =
+    MonitoringServerInterceptor(Clock.systemDefaultZone)(CFG)
+
+}
+
+case class MonitoringServerCall[Req, Res](
+    serverCall: ServerCall[Req, Res],
+    clock: Clock,
+    serverMetrics: ServerMetricsForMethod)(implicit CFG: Configuration)
+    extends ForwardingServerCall.SimpleForwardingServerCall[Req, Res](serverCall) {
+
+  private[this] val millisPerSecond = 1000L
+
+  private[this] val startInstant: Instant = clock.instant
+  reportStartMetrics()
```

Comment:
Is this call to `reportStartMetrics()` intentional? in that case, I think we should add a blank line between the call and the private val